parser/src/main/java/com/linkedin/avroutil1/model/AvroUnionSchema.java

```diff
@@ -21,7 +21,7 @@
     private List<SchemaOrRef> types;
 
     public AvroUnionSchema(CodeLocation codeLocation) {
-        super(codeLocation);
+        super(codeLocation, null); //cant put extra props on a union
```

Comment:
There are no null checks in the base class (AvroSchema) when accessing props-related methods (propertyNames(), getPropertyAsObject(), etc). I'm not sure if those methods can be called under _some_ callpath for unions; but to be safe, maybe pass in an JsonPropertiesContainer.EMPTY instead of null?